File: 0001_two_sum/go/solution.go
package two_sum

func twoSum(nums []int, target int) []int {
	seen := make(map[int]int, len(nums))
	for i, num := range nums {
		complement := target - num
		if j, exists := seen[complement]; exists {
			return []int{j, i}
		}
		seen[num] = i
	}
	return nil
}

File: 0001_two_sum/kotlin/solution.kt
fun twoSum(nums: IntArray, target: Int): IntArray {
    val seen = mutableMapOf<Int, Int>()

    for ((index, num) in nums.withIndex()) {
        val complement = target - num
        val complementIndex = seen[complement]
        if (complementIndex != null) {
            return intArrayOf(complementIndex, index)
        }
        seen[num] = index
    }

    return intArrayOf()
}
